custom-services/custom-service.md

```diff
@@ -0,0 +1,57 @@
+
+
+# Terminology
+
+- **custom service:** This refers to a running service (API Server) that is accessible via a hostname and provides custom business logic.
+- **mobile client application:** An instance of an application running on a mobile device.
+- **mobile enabled:** A service that has the required objects (configmap and secret) created along with the label "mobile:enabled".
+
+# Problem
+
+Currently there is no simple way for a mobile application developer using OpenShift or Kubernetes to expose the configuration details required to access a custom service running in the cluster to a mobile client application. 
+It would require knowledge of configmaps, secrets and our labeling schema to achieve this.
+While it would be something a developer familiar with OpenShift/K8s could do, we can provide a better user experience.
+Currently we allow configuration to be exposed for the mobile enabled services such as Keycloak that we have created specific APBs for, but most real world mobile client applications require some level of custom business logic. It is important that we allow this business logic to be exposed by the mobile developer to their mobile client applications in a consistent intuitive and simple fashion. If possible once the custom service has been created it should be viewed in the UI as just another mobile service.
+
+## Use cases we hope to solve
+- As a mobile developer, I want to expose a service I have already provisioned from the catalog to my mobile clients, so that I can make use of this service in a consistent and simple fashion.
+
+- As a mobile developer, there is a service with custom logic running somewhere that I need to make use of from within my client, I want to easily make this service accessible to my client devices with the needed configuration so that I can consume the custom business logic within my mobile client application.
+
+# Proposed Solution
+
+### Add enable mobile button / link
+To allow a developer to expose a service already provisioned from the service catalog, we could modify the UI view in OpenShift to have a call
+to action (something like) "enable mobile" that would show up on the overview screen inside the ServiceInstance box if mobile was enabled within the cluster and there was at least one mobile client in the namespace. When clicked they would be presented with a modal form that would allow them to specify needed configuration details (url, API Keys, any other key value pairs needed in order to consume the service). The key value pairs would be dynamic (as in they could add as many as needed) there is already options for this type of field in the OpenShift UI (used to specify env vars).
+When saved we would create the required objects (ie the configmap and a secret if needed) in the right format and with the correct labels to allow it to be consumed from the mobile CLI.
```

Comment:
Could probably link to the video from the POC that shows the 'External Service' dialog